pkg/kapp/resources/mod_field_remove.go

```diff
@@ -96,6 +96,19 @@ func (t FieldRemoveMod) apply(obj interface{}, path Path) error {
 			default:
 				panic(fmt.Sprintf("Unknown array index: %#v", part.ArrayIndex))
 			}
+		case part.Regex != nil && part.Regex.Regex != nil:
+			matchedKeys, err := t.obtainMatchingRegexKeys(obj, *part.Regex.Regex)
```

Comment:
I guess we will need the source object for remove as well?

Reply:
As `sources` in rebaseRules is only used with `copy` not with `remove` so that's not available and hence not required here.